src/OrleansServiceBus/Providers/Streams/EventHub/EventHubQueueCache.cs

```diff
@@ -115,14 +134,30 @@ public bool IsUnderPressure()
     /// </summary>
     public class EventHubQueueCache : EventHubQueueCache<CachedEventHubMessage>
     {
-        public EventHubQueueCache(IStreamQueueCheckpointer<string> checkpointer, IObjectPool<FixedSizeBuffer> bufferPool)
-            : this(checkpointer, new EventHubDataAdapter(bufferPool))
+        private readonly Logger log;
+
+        public EventHubQueueCache(IStreamQueueCheckpointer<string> checkpointer, IObjectPool<FixedSizeBuffer> bufferPool, Logger logger)
+            : this(checkpointer, new EventHubDataAdapter(bufferPool), logger)
+        {
+        }
+
+        public EventHubQueueCache(IStreamQueueCheckpointer<string> checkpointer, ICacheDataAdapter<EventData, CachedEventHubMessage> cacheDataAdapter, Logger logger)
+            : base(EventHubAdapterReceiver.MaxMessagesPerRead, checkpointer, cacheDataAdapter, EventHubDataComparer.Instance, logger)
         {
+            log = logger.GetSubLogger("-ehcache");
         }
 
-        public EventHubQueueCache(IStreamQueueCheckpointer<string> checkpointer, ICacheDataAdapter<EventData, CachedEventHubMessage> cacheDataAdapter)
-            : base(EventHubAdapterReceiver.MaxMessagesPerRead, checkpointer, cacheDataAdapter, EventHubDataComparer.Instance)
+        /// <summary>
+        /// Handles cache purge signals
+        /// </summary>
+        /// <param name="lastItemPurged"></param>
+        /// <param name="newestItem"></param>
+        protected override void OnPurge(CachedEventHubMessage? lastItemPurged, CachedEventHubMessage? newestItem)
         {
+            if (lastItemPurged.HasValue && newestItem.HasValue)
+            {
+                log.Info($"CachePeriod: EnqueueTimeUtc: {LogFormatter.PrintDate(lastItemPurged.Value.EnqueueTimeUtc)} to {LogFormatter.PrintDate(newestItem.Value.EnqueueTimeUtc)}, DequeueTimeUtc: {LogFormatter.PrintDate(lastItemPurged.Value.DequeueTimeUtc)} to {LogFormatter.PrintDate(newestItem.Value.DequeueTimeUtc)}");
+            }
```

Comment:
I'd check log.IsInformational() first at least, as this looks like a relatively costly string concatenation